pkg/util/httprequest/httprequest.go

```diff
@@ -100,8 +85,6 @@ func SchemeHost(req *http.Request) (string /*scheme*/, string /*host*/) {
 		scheme = "https"
 	case len(req.URL.Scheme) > 0:
 		scheme = req.URL.Scheme
-	case port == "443":
```

Comment:
removing this is wrong... it means we can redirect to http when the request is expecting https